### PR TITLE
feat(matches): add filter on latest deadline for matchable senders

### PR DESCRIPTION
**Pull request by AdrianAndersen**

### CODE REVIEW

File: src/collections/match/operations/match-generate-operation-helper.ts [hunk 1]
```diff
@@ -66,6 +69,7 @@ export async function getMatchableSenders(
         "handoutInfo.handoutById": {
           $in: branchIds.map((branchId) => new ObjectId(branchId)),
         },
+        deadline: { $gt: new Date(), $lt: new Date(deadline) },
```
Comment (LarsSelbekk) [inline src/collections/match/operations/match-generate-operation-helper.ts:72]:
```suggestion
        deadline: { $gt: new Date(), $lte: new Date(deadline) },
```

File: src/collections/match/operations/match-generate-operation-helper.ts [hunk 2]
```diff
@@ -23,6 +23,7 @@ export interface MatcherSpec {
   standLocation: string;
   userMatchLocations: MatchLocation[];
   startTime: string;
+  deadline: string;
```
Comment (LarsSelbekk) [inline src/collections/match/operations/match-generate-operation-helper.ts:26]: Rename to `deadlineBefore`, `latestDeadlineCutoff`, `lastDeadline` or similar

**Review by LarsSelbekk — CHANGES_REQUESTED** [review]

**Comment by AdrianAndersen** [comment]
[Kun matche ordre/customerItems for dette semesteret](https://trello.com/c/0hDXx9kj/664-kun-matche-ordre-customeritems-for-dette-semesteret)

File: src/collections/match/operations/match-generate-operation-helper.ts (hunk 1 above)
Comment (AdrianAndersen) [reply src/collections/match/operations/match-generate-operation-helper.ts]: fixo

**Review by LarsSelbekk — APPROVED** [review]

**Pull request merged**